pkg/proxy/iptables/proxier.go

```diff
@@ -1407,6 +1407,10 @@ func (proxier *Proxier) syncProxyRules() {
 			} else {
 				// TODO: Make all nodePorts jump to the firewall chain.
 				// Currently we only create it for loadbalancers (#33586).
+
+				// Fix localhost martian source error
+				writeLine(proxier.natRules, append(args, "-d", "127.0.0.1", "-j", string(KubeMarkMasqChain))...)
```

Comment:
should be 127.0.0.0/8 probably ?

since we know we're going to DNAT later, maybe this should check `-s` instead of `-d`?  Just for humans sake - it should behave the same...

Reply:
Make sense. I just pushed an update.